lib/parse-template.js

```diff
@@ -2,9 +2,13 @@
 const Transform = require('./transform');
 
 module.exports = function parseTemplate (handledTags, insertBeforePipeTags) {
-    return (baseTemplate, childTemplate) => new Promise((resolve) => {
-        const transform = new Transform(handledTags, insertBeforePipeTags);
-        const serializedList = transform.applyTransforms(baseTemplate, childTemplate);
-        resolve(serializedList);
+    return (baseTemplate, childTemplate) => new Promise((resolve, reject) => {
+        try {
+            const transform = new Transform(handledTags, insertBeforePipeTags);
+            const serializedList = transform.applyTransforms(baseTemplate, childTemplate);
+            resolve(serializedList);
+        } catch (e) {
+            reject(e);
+        }
```

Comment:
Why not 

```js
return () => Promise.resolve(new Transform(a,b)).then(transform => transform.applyTransforms(c,d));
```

here ... `new Transform(a,b)` should not throw ... if it did, it's a library error and we should fix it ... if `applyTransforms` throws then it's a lib-user error, and he might choose to handle it at the end of the chain.